bundles/org.openhab.binding.tivo/src/main/resources/OH-INF/thing/thing-types.xml

```diff
@@ -0,0 +1,139 @@
+					connection to the TiVo open. Recommended for monitoring the TiVo for changes in TV channels. Disable if other
+					applications that use the Remote Control Protocol port will also be used e.g. mobile remote control applications.<![CDATA[ <b> ]]>Default:
+					True (Enabled)<![CDATA[ </b> ]]></description>
+			</parameter>
+			<parameter name="pollForChanges" type="boolean" required="true">
+				<default>true</default>
+				<label>Poll for Channel Changes</label>
+				<description>Check
+					TiVo for channel changes. Enable if openHAB and a physical remote control (or other services use the Remote Control
+					Protocol) will be used. <![CDATA[ <b> ]]>Default:
+					True (Enabled)<![CDATA[ </b> ]]></description>
+			</parameter>
+			<parameter name="pollInterval" type="integer" max="65535" min="3" required="true">
+				<default>10</default>
+				<label>Polling Interval (Seconds)</label>
+				<description>Number
+					of seconds between polling jobs to update status information from the TiVo. <![CDATA[ <b> ]]>Default:
+					10<![CDATA[ </b> ]]></description>
+			</parameter>
+			<parameter name="cmdWaitInterval" type="integer" max="500" min="0" required="true">
+				<default>200</default>
+				<label>Command Wait Interval (Milliseconds)</label>
+				<description>Period
+					to wait AFTER a command is sent to the TiVo in milliseconds, before checking that the command has completed. <![CDATA[ <b> ]]>Default:
+					200<![CDATA[ </b> ]]></description>
+				<advanced>true</advanced>
+			</parameter>
+		</config-description>
+	</thing-type>
+
+	<channel-type id="channelForce">
+		<item-type>Number</item-type>
+		<label>Current Channel - Forced (FORCECH)</label>
+		<description>Displays the current channel number. When changed, tunes the DVR to the specified channel, cancelling any
+			recordings in progress if necessary i.e. all tuners are already in use / recording. The TiVo must be in Live TV mode
+			for this command to work. Type: Number (1-9999) [Decimals allowed for OTA sub-channels], DisplayFormat: %d</description>
+		<state min="1" max="9999" step="0.1" pattern="%d"></state>
+	</channel-type>
+	<channel-type id="channelSet">
+		<item-type>Number</item-type>
+		<label>Current Channel - Request (SETCH)</label>
+		<description>Displays the current channel number. When changed, tunes the DVR to the specified channel (unless a
+			recording is in progress on all available tuners). The TiVo must be in Live TV mode for this command to work. Type:
+			Number (1-9999) [Decimals allowed for OTA sub-channels], DisplayFormat: %d</description>
+		<state min="1" max="9999" step="0.1" pattern="%d"/>
+	</channel-type>
+	<channel-type id="isRecording">
+		<item-type>Switch</item-type>
+		<label>Recording</label>
+		<description>Indicates if the current channel is recording.</description>
+		<state readOnly="true"/>
+	</channel-type>
+	<channel-type id="menuTeleport">
+		<item-type>String</item-type>
+		<label>Change Special/Menu Screen (TELEPORT)</label>
```

Comment:
Labels are expected to be as short as possible. Guideline is 2-3 words with up to 25 chars. See https://www.openhab.org/docs/developer/bindings/thing-xml.html#formatting-labels-and-descriptions